### PR TITLE
WFLY-12005 Add documentation section for tuning distributed web application performance

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-12005

### CODE REVIEW

File: docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc [hunk 1]
```diff
@@ -195,3 +195,132 @@ e.g.
...
+
+One of the primary design goals of WildFly's distributed session manager was the parity of HttpSession semantics between distributable and non-distributable web applications.
+In order to provide predictable behavior suitable for most web applications, the default distributed session manager configuration is quite conservative, generally favoring consistency over availability.
+However, these defaults may not be appropriate for your application.
+In general, the effective performance of the distributed session manager is constrained by:
+
+. Replication/persistence payload size
+. Locking/isolation of a given session
+
+To optimize the configuration of the distributed session manager for your application, you can address the above constraints tuning one or more of the following:
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc]: s/tuning/by tuning/g

File: docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc [hunk 2]
```diff
@@ -195,3 +195,132 @@ e.g.
...
+* <<session_attribute_marshalling,Marshalling>>
+
+[[session_granularity]]
+=== Session granularity
+
+By default, WildFly's distributed session manager uses SESSION granularity, meaning that all session attributes are stored within a single cache entry.
+While this ensures that any object references shared between session attributes are preserved following replication/persistence, it means that a change to a single attribute results in the replication/persistence of *all* attributes.
+
+If your application does not share any object references between attributes, users are strongly advised to use ATTRIBUTE granularity.
+Using ATTRIBUTE granularity, each session attribute is stored a separate cache entry.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc]: s/stored a/stored in a/g

**Review by bstansberry — CHANGES_REQUESTED** [review]
Good stuff. I noted a couple trivial flaws. Please fell free to merge.

File: docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc (hunk 2 above)
Comment (pferraro) [reply docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc]: Done

File: docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc (hunk 1 above)
Comment (pferraro) [reply docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc]: Done

**Pull request merged**